sql/moz-fx-data-shared-prod/mozilla_vpn_derived/subscription_events_live/view.sql

```diff
@@ -1,89 +1,124 @@
 CREATE OR REPLACE VIEW
   `moz-fx-data-shared-prod.mozilla_vpn_derived.subscription_events_live`
 AS
-WITH json_arrays AS (
+WITH all_subscriptions AS (
   SELECT
     *,
     TO_JSON_STRING(promotion_codes) AS json_promotion_codes
   FROM
-    `moz-fx-data-shared-prod`.mozilla_vpn_derived.all_subscriptions_v1
+    mozdata.mozilla_vpn.all_subscriptions
+),
+max_active_date AS (
+  SELECT AS VALUE
+    MAX(active_date)
+  FROM
+    mozdata.mozilla_vpn.active_subscription_ids
+),
+new_events AS (
+  SELECT
+    active_date AS event_date,
+    subscription_id,
+    "New" AS event_type,
+  FROM
+    mozdata.mozilla_vpn.active_subscription_ids
+  WHERE
+    TRUE -- zetasql requires QUALIFY to be used in conjunction with WHERE, GROUP BY, or HAVING
+  QUALIFY
+    LAG(active_date) OVER (PARTITION BY subscription_id ORDER BY active_date) IS DISTINCT FROM (
+      active_date - 1
+    )
+),
+cancelled_events AS (
+  SELECT
+    active_date AS event_date,
+    subscription_id,
+    "Cancelled" AS event_type,
+  FROM
+    mozdata.mozilla_vpn.active_subscription_ids
+  CROSS JOIN
+    max_active_date
+  WHERE
+    active_date < max_active_date
+  QUALIFY
+    LEAD(active_date) OVER (PARTITION BY subscription_id ORDER BY active_date) IS DISTINCT FROM (
+      active_date + 1
+    )
+),
+events AS (
+  SELECT
+    *
+  FROM
+    new_events
+  UNION ALL
+  SELECT
+    *
+  FROM
+    cancelled_events
 )
 SELECT
-  event_date,
-  event_type,
-  granular_event_type,
-  plan_id,
-  status,
-  country,
-  country_name,
-  entrypoint_experiment,
-  entrypoint_variation,
-  utm_campaign,
-  utm_content,
-  utm_medium,
-  utm_source,
-  utm_term,
-  provider,
-  plan_amount,
-  billing_scheme,
-  plan_currency,
-  plan_interval,
-  plan_interval_count,
-  product_id,
-  product_name,
-  pricing_plan,
-  normalized_acquisition_channel,
-  normalized_campaign,
-  normalized_content,
-  normalized_medium,
-  normalized_source,
-  website_channel_group,
-  JSON_VALUE_ARRAY(json_promotion_codes) AS promotion_codes,
+  events.event_date,
+  events.event_type,
+  CASE
+  -- "New" event.subscription_ids
```

Comment:
Accidental search/replace?
```suggestion
  -- "New" events
```